File: eCommerce/eCommerce.Services/Database/Cupon.cs

using eCommerce.Model.Enums;
using System.ComponentModel.DataAnnotations;

namespace eCommerce.Services.Database
{
    public class Cupon
    {
        [Key]
        public int Id { get; set; }
        public string Code { get; set; } = string.Empty;
        public double DiscountAmount { get; set; }
        public DiscountType DiscountType { get; set; }
        public int Uses { get; set; }
        public DateTime ExpiresAt { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation property for orders using this coupon
        public ICollection<Order> Orders { get; set; } = new List<Order>();
    }
}

File: eCommerce/UI/ecommerce_desktop/lib/models/cupon.dart
import 'package:ecommerce_desktop/models/discount_type_enum.dart';
import 'package:json_annotation/json_annotation.dart';

part 'cupon.g.dart';

@JsonSerializable()
class Cupon {
  final int id;
  final String code;
  final double discountAmount;
  final DiscountType discountType;
  final int uses;
  final DateTime expiresAt;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Cupon({
    required this.id,
    required this.code,
    required this.discountAmount,
    required this.discountType,
    required this.uses,
    required this.expiresAt,
    required this.isActive,
    required this.createdAt,
    this.updatedAt,
  });

  factory Cupon.fromJson(Map<String, dynamic> json) => _$CuponFromJson(json);

  Map<String, dynamic> toJson() => _$CuponToJson(this);
}
